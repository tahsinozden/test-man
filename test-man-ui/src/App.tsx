import React, {useEffect, useState} from 'react';
import './App.css';

import Container from 'react-bootstrap/Container';
import TestPreviewList from "./components/TestPreviewList";
import CreateTestModal from "./components/CreateTestModal";
import {Col, Row} from "react-bootstrap";


const App: React.FC = () => {
    const [allTests, setAllTests] = useState([]);

    const loadTestDetails = () => {
        const requestOptions = {
            method: 'GET',
            headers: {'Content-Type': 'application/json'},
        };
        // TODO: get URL from configs
        fetch(`http://localhost:8080/api/v1/tests`, requestOptions)
            .then(response => response.json())
            .then(data => setAllTests(data));
    }

    useEffect(() => {
        loadTestDetails()
    }, []);
    return (
        <>
            <Container className="p-3">
                <Row>
                    <Col>
                        <CreateTestModal/>
                    </Col>
                </Row>
            </Container>
            <Container className="p-3">
                <Row>
                    <TestPreviewList testDetails={allTests} onTestDataChange={loadTestDetails}/>
                </Row>
            </Container>
        </>
    );
};

export default App;
