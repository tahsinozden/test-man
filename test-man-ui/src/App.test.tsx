import React from 'react';
import renderer from "react-test-renderer";
import App from "./App";

window.matchMedia = window.matchMedia || function () {
    return {
        matches: false,
        addListener: function () {
        },
        removeListener: function () {
        }
    };
};

it('renders correctly', () => {
    const tree = renderer
        .create(<App/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});