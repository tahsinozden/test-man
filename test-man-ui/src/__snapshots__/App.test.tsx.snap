// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-light bg-light"
  >
    <div
      className="container"
    >
      <span
        className="navbar-brand"
      >
        Test Manager
      </span>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div
      className="card"
    >
      <div
        className="p-3 container"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Create a new test
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="container"
      >
        <h5>
          Tests
        </h5>
        <div
          className="p-3 container"
        >
          <div
            className="row"
          >
            <div
              className="container"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
